include/nbl/builtin/hlsl/mpl.hlsl

```diff
@@ -55,9 +81,40 @@ struct log2
     static const uint16_t value = X ? (1ull<<6)-clz<X>::value-1 : -1ull;
 };
 
-#endif
 
+template<class T, class U = T>
+T bit_cast(U val);
+
+#define NBL_DECLARE_BIT_CAST(FROM, TO) template<> TO bit_cast<TO, FROM>(FROM val) { return impl::as##TO (val); }
+
+#define NBL_DECLARE_BIT_CAST_TYPES(BASE, BITS) \
+    NBL_DECLARE_BIT_CAST(BASE ## BITS, float ## BITS) \
+    NBL_DECLARE_BIT_CAST(BASE ## BITS, uint ## BITS) \
+    NBL_DECLARE_BIT_CAST(BASE ## BITS, int ## BITS)
+
+#define NBL_DECLARE_BIT_CAST_BITS(BASE) \
+    NBL_DECLARE_BIT_CAST_TYPES(BASE, 16_t) \
+    NBL_DECLARE_BIT_CAST_TYPES(BASE, 32_t) \
+    NBL_DECLARE_BIT_CAST_TYPES(BASE, 64_t)
+
+NBL_DECLARE_BIT_CAST_BITS(float)
+NBL_DECLARE_BIT_CAST_BITS(uint)
+NBL_DECLARE_BIT_CAST_BITS(int)
+
+#undef NBL_DECLARE_BIT_CAST
+#undef NBL_DECLARE_BIT_CAST_TYPES
+#undef NBL_DECLARE_BIT_CAST_BITS
+
+#else
+
+template<class T, class U>
+constexpr T bit_cast(const U& val)
+{
+    return std::bit_cast<T, U>(val);
 }
+
+#endif
```

Comment:
why is bitcast in `mpl`? it should be in `bit.hlsl` and not inside any namespace!